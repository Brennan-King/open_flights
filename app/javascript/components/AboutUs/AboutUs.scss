.about-us-container {
  text-align: center;
}

.about-us-header {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.about-us-paragraph {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: large;
  margin-left: 400px;
  margin-right: 400px
}

.about-us-image {
  border-radius: 30px;
  height: 500px;
  width: 750px;
  padding: 15px;
}