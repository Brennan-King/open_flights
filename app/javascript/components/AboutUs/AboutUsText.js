/* eslint-disable max-len */
export const aboutUsParagraphOneText = 'OpenFlight\'s mission is to provide reliable user reviews that are unbiased and trustworthy. We hope to provide users with feedback on major airlines to ensure they feel comfortable with the airline they select. Browse reviews for dozens of airlines accross the world.';

export const aboutUsParagragphTwoText = 'Have any experiences with a specific airline? Feel free to leave a review and share how you feel with other users.';
