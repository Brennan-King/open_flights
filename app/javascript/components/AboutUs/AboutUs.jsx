import React from 'react';
import aboutUsImageOne from '../../images/aboutUsImageOne.jpg';
import aboutUsImageTwo from '../../images/aboutUsImageTwo.jpeg';
import aboutUsImageThree from '../../images/aboutUsImageThree.jpg';
import {aboutUsParagraphOneText, aboutUsParagragphTwoText} from './AboutUsText';
import './AboutUs.scss';

/**
 * Constructs a static about us page.
 * @return {node} a div containing the configured about us page
 */
const AboutUs = () => {
  return (
    <div className="about-us-container">
      <h1 className="about-us-header">Our Mission</h1>
      <p className="about-us-paragraph">{aboutUsParagraphOneText}</p>
      <div>
        <img
          className="about-us-image"
          src={aboutUsImageOne}
          alt="aboutUsAirplane"
        />
        <img
          className="about-us-image"
          src={aboutUsImageTwo}
          alt="aboutUsAirplane"
        />
      </div>
      <p className="about-us-paragraph">{aboutUsParagragphTwoText}</p>
      <div>
        <img
          className="about-us-image"
          src={aboutUsImageThree}
          alt="aboutUsAirplane"
        />
      </div>
    </div>
  );
};

export default AboutUs;
