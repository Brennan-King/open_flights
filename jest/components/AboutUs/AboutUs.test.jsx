/* eslint-disable max-len */
import React from 'react';
import AboutUs from '../../../app/javascript/components/AboutUs/AboutUs';
import renderer from 'react-test-renderer';

/**
 * Test suite for AboutUs component
 */
describe('AboutUs', () => {
  it('should render an AboutUs component', () => {
    const component = renderer.create(<AboutUs />);
    const aboutUs = component.toJSON();

    expect(aboutUs).toMatchSnapshot();
  });
});
