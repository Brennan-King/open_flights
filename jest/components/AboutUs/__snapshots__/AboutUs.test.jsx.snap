// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs should render an AboutUs component 1`] = `
<div
  className="about-us-container"
>
  <h1
    className="about-us-header"
  >
    Our Mission
  </h1>
  <p
    className="about-us-paragraph"
  >
    OpenFlight's mission is to provide reliable user reviews that are unbiased and trustworthy. We hope to provide users with feedback on major airlines to ensure they feel comfortable with the airline they select. Browse reviews for dozens of airlines accross the world.
  </p>
  <div>
    <img
      alt="aboutUsAirplane"
      className="about-us-image"
      src="test-file-stub"
    />
    <img
      alt="aboutUsAirplane"
      className="about-us-image"
      src="test-file-stub"
    />
  </div>
  <p
    className="about-us-paragraph"
  >
    Have any experiences with a specific airline? Feel free to leave a review and share how you feel with other users.
  </p>
  <div>
    <img
      alt="aboutUsAirplane"
      className="about-us-image"
      src="test-file-stub"
    />
  </div>
</div>
`;
